indexing-service/src/main/java/org/apache/druid/indexing/input/DruidSegmentInputFormat.java

```diff
@@ -55,14 +55,27 @@ public InputEntityReader createReader(
       File temporaryDirectory
   )
   {
-    return new DruidSegmentReader(
-        source,
-        indexIO,
-        inputRowSchema.getTimestampSpec(),
-        inputRowSchema.getDimensionsSpec(),
-        inputRowSchema.getColumnsFilter(),
-        dimFilter,
-        temporaryDirectory
-    );
+    // this method handles the case when the entity comes from a tombstone or from a regular segment
+
+    InputEntityReader retVal = null;
+    // source is tombstone case
+    if (source instanceof DruidSegmentInputEntity) {
```

Comment:
Is there any tests that the correct segmentReader is created?

Reply:
This should never be called with another type of entity. I added `    Preconditions.checkArgument(source instanceof DruidSegmentInputEntity);` to catch that otherwise.

Reply:
Also added a unit test for that class to test that reader since there was no unit test, good call.